object/lock.go

```diff
@@ -13,21 +13,20 @@ import (
 type Lock v2object.Lock
 
 // WriteLock writes Lock to the Object, and sets its type to TypeLock.
-// The object must not be nil.
 //
 // See also ReadLock.
-func WriteLock(obj *Object, l Lock) {
-	obj.SetType(TypeLock)
-	obj.SetPayload(l.Marshal())
+func (o *Object) WriteLock(l Lock) {
+	o.SetType(TypeLock)
+	o.SetPayload(l.Marshal())
 }
 
 // ReadLock reads Lock from the Object. The lock must not be nil.
 // Returns an error describing incorrect format. Makes sense only
 // if object has TypeLock type.
 //
 // See also WriteLock.
-func ReadLock(l *Lock, obj Object) error {
-	return l.Unmarshal(obj.Payload())
+func (o Object) ReadLock(l *Lock) error {
```

Comment:
Non-pointer receiver here? Makes it different from others.

Reply:
I tried to follow the object using in the original function. But, yeah, it should be similar. Updated to pointer